pkg/api/handlers/libpod/images.go

```diff
@@ -498,12 +502,74 @@ func CommitContainer(w http.ResponseWriter, r *http.Request) {
 	if len(query.Repo) > 0 {
 		destImage = fmt.Sprintf("%s:%s", query.Repo, tag)
 	}
-	commitImage, err := ctr.Commit(r.Context(), destImage, options)
+
+	if !query.Stream {
+		commitImage, err := ctr.Commit(r.Context(), destImage, options)
+		if err != nil && !strings.Contains(err.Error(), "is not running") {
+			utils.Error(w, http.StatusInternalServerError, fmt.Errorf("CommitFailure: %w", err))
+			return
+		}
+		utils.WriteResponse(w, http.StatusOK, entities.IDResponse{ID: commitImage.ID()})
+		return
+	}
+
+	// Channels all mux'ed in select{} below to follow API commit protocol
+	stdout := channel.NewWriter(make(chan []byte))
+	defer stdout.Close()
+	// Channels all mux'ed in select{} below to follow API commit protocol
+	options.CommitOptions.ReportWriter = stdout
+	var commitImage *libimage.Image
+	success := false
+	runCtx, cancel := context.WithCancel(r.Context())
+	go func() {
+		defer cancel()
+		commitImage, err = ctr.Commit(r.Context(), destImage, options)
+		if err == nil {
+			success = true
+		}
+	}()
+
 	if err != nil && !strings.Contains(err.Error(), "is not running") {
 		utils.Error(w, http.StatusInternalServerError, fmt.Errorf("CommitFailure: %w", err))
 		return
 	}
-	utils.WriteResponse(w, http.StatusOK, entities.IDResponse{ID: commitImage.ID()})
+
+	flush := func() {
+		if flusher, ok := w.(http.Flusher); ok {
+			flusher.Flush()
+		}
+	}
+	body := w.(io.Writer)
+	enc := json.NewEncoder(body)
+
+	for {
+		type BuildResponse struct {
+			Stream string `json:"stream,omitempty"`
+		}
+		m := BuildResponse{}
+
+		select {
+		case e := <-stdout.Chan():
+			m.Stream = string(e)
```

Comment:
On first stream, we should set content-type to text/json

Reply:
How do I do that?